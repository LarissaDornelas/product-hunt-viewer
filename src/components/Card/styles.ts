import styled from "styled-components";
import { theme } from "styles/theme";

export const Container = styled.div`
  width: 100%;
  background-color: ${theme.colors.white};
  margin-top: 20px;
  border-radius: 5px;
  z-index: 1;
`;
