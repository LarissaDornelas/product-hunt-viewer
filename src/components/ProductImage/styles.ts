import styled from "styled-components";

import { ImageSize } from "./types";

export const Container = styled.div<{ imageUrl: string; size: ImageSize }>`
  background-image: url(${({ imageUrl }) => (imageUrl ? imageUrl : "")});
  background-repeat: no-repeat;
  background-size: cover;
  height: ${({ size }) => (size ? size : "40px")};
  width: ${({ size }) => (size ? size : "40px")};
  border-radius: 10px;
`;
