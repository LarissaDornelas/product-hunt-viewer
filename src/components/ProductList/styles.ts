import styled from "styled-components";

const Container = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  padding: 20px;
  margin: 110px 200px 0 200px;

  @media (max-width: 800px) {
    margin: 110px 0 0 0;
    max-width: 340px;
  }
`;

const LoadingContainer = styled.div`
  display: flex;
  padding: 30px;
  height: 30px;
  justify-content: center;
  align-items: center;
`;
const ObservableContainer = styled.div`
  display: flex;
  height: 5px;
  justify-content: center;
  align-items: center;
`;

export { Container, LoadingContainer, ObservableContainer };
