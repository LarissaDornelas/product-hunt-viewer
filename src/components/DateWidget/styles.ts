import styled from "styled-components";
import { theme } from "styles/theme";

const Container = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 200px;
  background-color: ${theme.colors.primary};
  height: 30px;
  border-radius: 10px;

  @media (max-width: 800px) {
    width: 150px;
  }

  p {
    font-size: 0.825rem;
  }
`;

export { Container };
