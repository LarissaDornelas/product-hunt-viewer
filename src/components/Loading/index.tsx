import React from "react";
import { Container } from "./styles";

const Loading: React.FC = () => {
  return <Container data-testid="loading-container" />;
};

export { Loading };
