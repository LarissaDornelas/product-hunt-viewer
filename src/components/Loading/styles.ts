import styled, { keyframes } from "styled-components";
import { theme } from "styles/theme";

const spin = keyframes`
    from{
        transform: rotate(0deg);
    }
    to{
        transform: rotate(360deg);
    }

`;

export const Container = styled.div`
  border: 5px solid ${theme.colors.shadow};
  border-top: 5px solid ${theme.colors.secondary};
  border-radius: 50%;
  width: 30px;
  height: 30px;
  animation: ${spin} 2s linear infinite;
`;
