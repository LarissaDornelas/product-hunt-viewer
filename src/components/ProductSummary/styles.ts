import styled from "styled-components";
import { theme } from "styles/theme";

const Container = styled.div`
  display: flex;
  position: relative;
  width: 100%;
  height: 100%;
  justify-content: space-between;
  align-items: center;
  padding: 5px 10px;
  height: 70px;
`;

const DataContainer = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  flex: 2;
  overflow: hidden;
  margin: 0px 100px 0px 15px;
  text-overflow: ellipsis;
  h1 {
    font-size: 0.9rem;
    padding: 0px;
    margin: 0px 0px 5px 0px;
    font-weight: 400;
  }

  p {
    font-size: 0.825rem;
    padding: 0px;
    margin: 0px;
    white-space: nowrap;
    max-width: 100px;
    color: ${theme.colors.fontPrimary};
  }
`;

export { Container, DataContainer };
