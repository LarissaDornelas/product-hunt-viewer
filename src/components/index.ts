import { ProfileImage } from "./ProfileImage";
import { DateWidget } from "./DateWidget";
import { SearchButton } from "./SearchButton";
import { Tab } from "./Tab";
import { Card } from "./Card";

export { ProfileImage, DateWidget, SearchButton, Tab, Card };
