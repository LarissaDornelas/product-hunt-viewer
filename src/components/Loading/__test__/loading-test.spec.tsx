import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";

import { Loading } from "components";
import { theme } from "styles/theme";

describe("Component: Loading", () => {
  afterEach(cleanup);

  it("Should render without crashing", () => {
    const container = document.createElement("div");
    ReactDOM.render(<Loading />, container);
  });

  it("Should render Loading correctly", () => {
    const { getByTestId } = render(<Loading />);

    expect(getByTestId("loading-container")).toHaveStyle({
      border: `5px solid ${theme.colors.shadow}`,
      borderTop: `5px solid ${theme.colors.secondary}`,
      borderRadius: "50%",
      width: "30px",
      height: "30px",
      animation: "bgrRMf 2s linear infinite",
    });
  });

  it("matches snapshot", () => {
    const tree = renderer.create(<Loading />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
