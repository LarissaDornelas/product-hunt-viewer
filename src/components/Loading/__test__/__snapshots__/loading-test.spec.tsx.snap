// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Loading matches snapshot 1`] = `
<div
  className="sc-dQppl iutGYG"
  data-testid="loading-container"
/>
`;
